### PR TITLE
Minor detectives spawn with a candy cigarette and apple juice filled flask

### PR DESCRIPTION
**Pull request by NamelessFairy**
## About The Pull Request

If a detective joins who is 20 or younger their cigarette is changed for a candy one and their flask is filled with apple juice. Also adds candy cigarettes' to the detective vendor.

## Why It's Good For The Game

Minor crew members are unable to use cigarette vendors or acquire alcohol in game without effectively committing a crime, it doesn't make sense for minor detectives to spawn with them. I also think that minor characters trying to obtain narcotics can be a pretty entertaining RP starter as a overall harmless crime and something that requires interaction with other members of the crew.

## Changelog
:cl:
add: Minor detectives now spawn with a candy cigarette and a flask full of apple juice.
balance: Candy Cigarettes added to the detective vendor.
fix: Minor detectives no longer spawn with real cigarettes or real alcohol, you'll need to ask you colleagues to buy you some, hopefully they wont arrest you for trying.
/:cl:

### CODE REVIEW

**Comment by NamelessFairy** [comment]
I misclicked enter oh fuck

File: code/modules/vending/wardrobes.dm [hunk 1]
```diff
@@ -555,6 +555,7 @@
 		/obj/item/clothing/glasses/regular/kim = 1,
 		/obj/item/reagent_containers/cup/glass/flask/det = 2,
 		/obj/item/storage/fancy/cigarettes = 5,
+		/obj/item/storage/fancy/cigarettes/cigpack_candy = 5,
 		)
```
Comment (Fikou) [inline code/modules/vending/wardrobes.dm:559]:
```suggestion
	)
```

File: code/modules/jobs/job_types/detective.dm [hunk 2]
```diff
@@ -73,6 +73,12 @@
 		)
 	implants = list(/obj/item/implant/mindshield)
 
+/datum/outfit/job/detective/pre_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	. = ..()
+	if (H.age < AGE_MINOR)
```
Comment (Fikou) [inline code/modules/jobs/job_types/detective.dm:78]: use descript variable names

File: code/modules/clothing/head/jobs.dm [hunk 3]
```diff
@@ -117,6 +117,7 @@
 	icon_state = "detective"
 	inhand_icon_state = "det_hat"
 	var/candy_cooldown = 0
+	var/flask = /obj/item/reagent_containers/cup/glass/flask/det
```
Comment (Fikou) [inline code/modules/clothing/head/jobs.dm:120]: new vars at the bottom, needs documentation, change name to flask_path

**Review by Fikou — CHANGES_REQUESTED** [review]

**Comment by Bawhoppen** [comment]
https://www.youtube.com/watch?v=FhUixnJU2-Y

File: code/modules/clothing/head/jobs.dm [hunk 4]
```diff
@@ -118,6 +118,8 @@
 	inhand_icon_state = "det_hat"
 	var/candy_cooldown = 0
 	dog_fashion = /datum/dog_fashion/head/detective
+	//Path for the flask that spawns inside their hat roundstart
```
Comment (Fikou) [inline code/modules/clothing/head/jobs.dm:121]:
```suggestion
	///Path for the flask that spawns inside their hat roundstart
```
three slashes mean autodoc, two just mean a comment

**Review by Mothblocks — APPROVED** [review]
cute

**Comment by bluetshirtguy** [comment]
They deserve to be in jail for playing as a young detective tbh.

**Comment by Bawhoppen** [comment]
They should get a plastic "Honorary Detective" badge, like the kind kids get at county fairs or when the cops talk to the kids at school

**Comment by Shadowflame909** [comment]
Now this is the Soulful coding that makes this space station feel more alive

**Comment by JohnFulpWillard** [comment]
> They should get a plastic "Honorary Detective" badge, like the kind kids get at county fairs or when the cops talk to the kids at school

Detectives don't even have any badges currently (though they should so I can play as M. Vonel)

**Comment by Bawhoppen** [comment]
Instead of ID cards maybe they could get badges

**Comment by Livrah** [comment]
Replace .38 with a toy gun.

**Pull request merged**